src/_pytest/nodes.py

```diff
@@ -81,7 +88,7 @@ def __init__(
         self.config = config or parent.config
 
         #: the session this node is part of
-        self.session = session or parent.session
+        self.session = session or parent.session  # type: Session
```

Comment:
Would be better to type the `__init__` argument instead. Since it's `Optional[Session]`, the following might be needed, not sure:

```py
if session is None:
    assert parent.session is not None
    self.session = parent.session
else:
   self.session = session
```

Reply:
Seems to work with just "session: Session".

Reply:
Done.